sdks/python/apache_beam/transforms/cy_combiners.py

```diff
@@ -308,3 +308,94 @@ class AnyCombineFn(AccumulatorCombineFn):
 
 class AllCombineFn(AccumulatorCombineFn):
   _accumulator_type = AllAccumulator
+
+MAX_LONG_10_FOR_LEADING_ZEROS = [19, 18, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15,
+                                 15, 15, 15, 14, 14, 14, 13, 13, 13, 12, 12, 12,
+                                 12, 11, 11, 11, 10, 10, 10, 9, 9, 9, 9, 8, 8,
+                                 8, 7, 7, 7, 6, 6, 6, 6, 5, 5, 5, 4, 4, 4, 3, 3,
+                                 3, 3, 2, 2, 2, 1, 1, 1, 0, 0, 0]
+
+LONG_SIZE = 64
+
+def compare_to(x, y):
+  """return the sign bit of x-y"""
+  if x < y:
+    return 1
+  return 0
+
+def get_log10_round_to_floor(element):
+  number_of_leading_zeros = LONG_SIZE - element.bit_length()
+  y = MAX_LONG_10_FOR_LEADING_ZEROS[number_of_leading_zeros]
+  return y - compare_to(element, math.pow(10, y))
+
+class DistributionAccumulator(object):
+  """Distribution Counter:
+  contains value distribution statistics and methods for incrementing
+  """
+  def __init__(self):
+    self.min = 0
+    self.max = 0
+    self.count = 0
+    self.sum = 0
+    self.sum_of_squares = 0
+    """Histogram buckets of value counts for a distribution(1,2,5 bucketing)"""
```

Comment:
Hmm... did we want to hard code this bucketing? (Also seems worrisome that the buckets are not equally sized.)